hyperspy/events.py

```diff
@@ -90,22 +187,28 @@ def connect(self, function, nargs='all'):
         self._connected[nargs].add(function)
 
     def disconnect(self, function):
+        """
+        Disconnects a function from the event. The passed function will be
+        disconnected irregardless of which 'nargs' argument was passed to
+        connect().
+        """
         for c in self._connected.itervalues():
             if function in c:
                 c.remove(function)
 
-    def trigger(self, *args, **kwargs):
-        if not self.suppress:
-            for nargs, c in self._connected.iteritems():
+    def trigger(self, *args):
+        if not self._suppress:
+            # Loop on copy to deal with callbacks which change connections
+            for nargs, c in self._connected.copy().iteritems():
                 if nargs is 'all':
                     for f in c:
-                        f(*args, **kwargs)
```

Comment:
Well, `nargs` affects the number of `args` not `kwargs`. `args` are dangerous because it is easy to introduce bugs by getting their order wrong. If I had to choose I would ban `args` ;)